src/util-ip.c

```diff
@@ -65,7 +66,8 @@ bool IPv4AddressStringIsValid(const char *str)
 
     addr[dots][alen] = '\0';
     for (int x = 0; x < 4; x++) {
-        int a = atoi(addr[x]);
+        int a;
+        ByteExtractStringInt32(&a, 10, 0, (const char *)addr[x]);
```

Comment:
if ```ByteExtractStringInt32``` fails it may leave ```a``` untouched and thus the next check works on an uninitialized value. I've seen this pattern a couple of times in this PR, so please check all.